File: Greedy/chaechon/1931.py
# 회의실 배정

import sys 
from collections import deque

input = sys.stdin.readline
N = int(input())
meetings=[]
count = 0 
next = 0

for _ in range(N):
    a, b = map(int, input().split())
    meetings.append((a,b))

meetings.sort(key=lambda x: (x[1], x[0]))
# print(meetings)

for a, b in meetings:
    # print(a,b)
    if a>=next:
        # 회의 끝나는 시간보다 시작시간이 크면 시작 가능 -> 그 회의시간의 끝나는 시간을 다음 시작 가능한 시간으로 집어넣기
        count +=1
        next = b

print(count)

File: Greedy/jichon/1931.js
const fs = require("fs");
const input = fs.readFileSync("./input.txt").toString().trim().split("\n");

const solution = () => {
  let n = +input[0];
  let meetings = [];
  for (let i = 1; i < n + 1; i++) {
    let [start, end] = input[i].split(" ").map(Number);
    meetings.push({ start, end });
  }
  //끝나는 시간 기준으로 정렬 어케하지.. -> gpt 도움
  meetings.sort((a, b) => {
    if (a.end === b.end) return a.start - b.start;
    return a.end - b.end;
  });

  let count = 0;
  let lastEnd = 0;

  for (const { start, end } of meetings) {
    if (start >= lastEnd) {
      count++;
      lastEnd = end;
    }
  }
  console.log(count);
};

solution();
